packages/block-library/src/comments-query-loop/edit/comments-inspector-controls.js

```diff
@@ -21,10 +21,20 @@ const orderOptions = [
 	},
 ];
 
+const defaultPageOptions = [
+	{
+		label: __( 'Newest' ),
+		value: 'newest',
+	},
+	{
+		label: __( 'Oldest' ),
+		value: 'oldest',
+	},
+];
+
```

Comment:
Right now on Discussion Settings say:
Last or First page to display by default.

This could make confusion to the user, although, to be honest, Discussion settings are also confusing 😓 

cc: @jasmussen @jameskoster

Reply:
I have the same feeling. I ended using `oldest` and `newset` because those are the values used internally, but I'm not sure which option is clearer from the user's perspective. For me, it's like using `first` or `last` instead would be equivalent, although it doesn't give you information about what comments are going to be shown (`oldest` comments are in the `first` or the `last` page? 🤷 ).

Reply:
Took it for a spin:
![comments](https://user-images.githubusercontent.com/1204802/154221553-8bc83ee2-e72b-449b-a9ab-93efdcdd6a3d.gif)

I don't see any "Last or First page to display by default" on this page:

<img width="1088" alt="Screenshot 2022-02-16 at 09 04 03" src="https://user-images.githubusercontent.com/1204802/154221633-770a6168-546d-416d-9dde-ac308b5fa320.png">

Am I looking in the wrong place?

Right now it says "Default page", that seems okay to me. It's not super clear, but it's also a rare change to tweak. I'd be happy to start simple and then we can rephrase if it becomes necessary, for example to add help text.

Reply:
It's all very confusing because and I'm still not sure how it exactly works 🤣 

I think it's the following:
1. When you select the `last` page displayed by default then you sort all top-level comments by date where the newest comments are on the first page.
2. When you select the `older` comments at the top of each page, then it reverses the order of comments but only on individual pages.

So maybe we should have:
1. Order pages by...
2. Order items on the page by...

Reply:
Is it definitely necessary that a single comment loop can override the global discussion settings? 🤔  Comments are a fairly universal concept – doesn't it seem unlikely that a site would want a different treatment between post and page comments? 

I worry that this could encourage folks to customise all comment loops, at which point the global settings don't actually do anything. That could lead to some confusing UX. 

If we're committed to this change, should we update the language on the global discussion settings to clarify that the ordering and default page options are _defaults_, and might be overwritten by the theme / template?

Reply:
> I don't see any "Last or First page to display by default" on this page:

@jasmussen it's kind of hidden. 😅 🙈 

<img width="1023" alt="Screenshot 2022-02-16 at 11 28 38" src="https://user-images.githubusercontent.com/6917969/154246285-3001febb-6cfc-482a-9658-cad3d7cdfc18.png">

> So maybe we should have:
> 
> 1. Order pages by...
> 2. Order items on the page by...

Yup @gziolo, that would be more clear than we have now, it would be nice to have an option to change the order of pages as well. 🙂

AFAIK, WP settings don't allow you to change the _pages_ order, just the _items_ order on each page, which is what you select when you choose between _newer_ or _older_ comments at the top of each page. For that reason, the _first_ page was always the _oldest_ comments and the _last_ page the _newest_ ones.

I think it is clearer with this diagram:

![image](https://user-images.githubusercontent.com/6917969/154253080-4f3c88db-3c6e-4d1a-8283-39860beef993.png)

BTW, the option that is set internally when you choose between _newer_ or _older_ comments at the top of each page is [`comment_order`](https://codex.wordpress.org/Option_Reference), but the possible values for that option are `asc` or `desc` 😅, and can be confused with the `order` param you can pass to [`WP_Comment_Query`](https://developer.wordpress.org/reference/classes/wp_comment_query/__construct/), which also changes the pages order, so `comment_order` and `order` are not equivalent (I know, it's really confusing 😓).

Reply:
> Comments are a fairly universal concept – doesn't it seem unlikely that a site would want a different treatment between post and page comments?

You're right, although it would be great to also consider other cases, like adding a list of “latest comments” somewhere in your page, or “top comments”, or whatever…

Well, now I'm not sure if it would be better to create custom blocks for those specific cases. 🤔 

> If we're committed to this change, should we update the language on the global discussion settings to clarify that the ordering and default page options are _defaults_, and might be overwritten by the theme / template?

Wouldn't it be simpler if we warn in the block settings that you're overwriting the global discussion settings? Just asking, I don't have any strong opinion here. 😄

Reply:
> You're right, although it would be great to also consider other cases, like adding a list of “latest comments” somewhere in your page, or “top comments”, or whatever…

That's a good point, even held up against Jay's also excellent point. A few thoughts:

* Perhaps the toggle should be renamed "Inherit discussion settings", and we add help-text below that says something like "Use globally defined sort order and comment settings. You can configure these in [link]Discussion Settings[/link]".
* We start with less: perhaps we start with just "Items per page", and then instead of _two_ options for oldest vs. newest, we make a single toggle: "Show newest comments first", untoggled by default. We'd then use this setting for both "Order by" and "Default page". 

Keeping in mind we can always make it more granular as we go, it seems like this might also be an opportunity to simplify.

Reply:
> * [...] and we add help-text below that says something like "Use globally defined sort order and comment settings. You can configure these in [link]Discussion Settings[/link]".

That's great, I really like that suggestion. 😊 

> * [...] we make a single toggle: "Show newest comments first", untoggled by default. We'd then use this setting for both "Order by" and "Default page".

Yup, that would be simpler from the user perspective, and we can always add more features/complexity in the future.

I don't mind creating new PRs to address these changes, if we all agree on that. 👀

cc @gziolo @jameskoster @c4rl0sbr4v0 @michalczaplinski

Reply:
> Well, now I'm not sure if it would be better to create custom blocks for those specific cases.

Yes I considered things like Reviews in WooCommerce as well (which are just WP comments). They feel like variations of the comment query block to me. Similar to how the "Posts List" block is a variation of the Query block.

> Wouldn't it be simpler if we warn in the block settings that you're overwriting the global discussion settings?

I don't think so, from the UX perspective. To be clear I'm thinking of the other side of this flow. Consider an end user who installs a theme that elects not to inherit the global settings in each instance of the comment query loop block. When that user customises options in the global discussion settings, their changes will not be reflected on the frontend, and they will have no feedback explaining why.

Reply:
> Consider an end user who installs a theme that elects not to inherit the global settings in each instance of the comment query loop block. When that user customises options in the global discussion settings, their changes will not be reflected on the frontend, and they will have no feedback explaining why.

That's a good point, I haven't thought of that. 🤔 

I think it's better to continue this conversation in a different issue, I've created one: https://github.com/WordPress/gutenberg/issues/38864